### DISCUSSION TITLE
Suspense & HTML Streaming

### DISCUSSION THREAD

**Discussion by NickBlow** [preactjs/preact#3728] [General]
I searched the issues, so excuse me if this has been answered before. With React 18, React is able to use HTML streaming and Suspense to stream UI before a Suspense has been fulfilled. Are there any plans to support streaming / suspense in this manner? 

I'm aware this is a complex feature - but it does give the best of both worlds when it comes to SSR in my opinion.

**Comment by developit** [comment]
We have some explorations for this but haven't landed anything yet. It will only be a priority after v11 ships. 

Fwiw the technique in preact-iso provides 90% of the benefit of this without relying on a complicated out-of-order streaming and reparenting (it just uses suspense).
